File: C/Grace.c
#include <stdio.h>
#define YOLO
#define SWAG
#define FT(x)int main(){FILE *f=fopen(x, "w");char *c="#include <stdio.h>%c#define YOLO%c#define SWAG%c#define FT(x)int main(){FILE *f=fopen(x, %cw%c);char *c=%c%s%c;fprintf(f, c, 10, 10, 10, 34, 34, 34, c, 34, 10, 10, 9, 10, 10, 34, 34);fclose(f);}%c/*%c%cThis program will print its own source when run.%c*/%cFT(%c./Grace_kid.c%c)";fprintf(f, c, 10, 10, 10, 34, 34, 34, c, 34, 10, 10, 9, 10, 10, 34, 34);fclose(f);}
/*
	This program will print its own source when run.
*/
FT("./Grace_kid.c")

File: PHP/Grace.php
<?php
define("42", "42");
define("status", "blackhat");
define("github", "cheater");
/*
    This program will print its own source when run.
*/
$handle = fopen("Grace_kid.php", "w");
$str = '<?php%cdefine("42", "42");%cdefine("status", "blackhat");%cdefine("github", "cheater");%c/*%c    This program will print its own source when run.%c*/%c$handle = fopen("Grace_kid.php", "w");%c$str = %c%s%c;%cfprintf($handle, $str, 10, 10, 10, 10, 10, 10, 10, 10, 39, $str, 39, 10, 10, 10, 10, 10);%cfclose($handle);%c?>%c';
fprintf($handle, $str, 10, 10, 10, 10, 10, 10, 10, 10, 39, $str, 39, 10, 10, 10, 10, 10);
fclose($handle);
?>
